src/parser/parser_verification.js

```diff
@@ -0,0 +1,180 @@
+    }
+  } else if (parserUtils.parseNodeText(node).length === 0) {
+    emitMissingValueWarning(
+      { name: node.nodeName, parentName: node.parentNode.nodeName },
+      emit
+    );
+    return;
+  }
+}
+
+/**
+ * Verify and trigger warnings if node required attributes are not set.
+ * @param  {Node} node - The node element.
+ * @param  {Function} emit - Emit function used to trigger Warning event.
+ */
+function verifyRequiredAttributes(node, emit) {
+  if (
+    !requiredValues[node.nodeName] ||
+    !requiredValues[node.nodeName].attributes
+  ) {
+    return;
+  }
+  const requiredAttributes = requiredValues[node.nodeName].attributes;
+  const missingAttributes = [];
+  requiredAttributes.forEach(attributeName => {
+    const attribute = node.getAttribute(attributeName);
+    if (!attribute) {
+      missingAttributes.push(attributeName);
+    }
+  });
+  if (missingAttributes.length > 0) {
+    emitMissingValueWarning(
+      {
+        name: node.nodeName,
+        parentName: node.parentNode.nodeName,
+        attributes: missingAttributes
+      },
+      emit
+    );
+  }
+}
+
+/**
+ * Verify and trigger warnings if node required sub element are not set.
+ * @param  {Node} node - The node element
+ * @param  {Boolean} isAdInline - True if node is contained in a inline
+ * @param  {Function} emit - Emit function used to trigger Warning event.
+ */
+function verifyRequiredSubElements(node, emit, isAdInline) {
+  const required = requiredValues[node.nodeName];
+
+  if (!required) {
+    return;
+  }
+
+  if (required.subElements) {
+    const requiredSubElements = required.subElements;
+    const missingSubElements = [];
+
+    requiredSubElements.forEach(subElementName => {
+      const subElement = parserUtils.childByName(node, subElementName);
+      if (!subElement) {
+        missingSubElements.push(subElementName);
+      }
+    });
+
+    if (missingSubElements.length > 0) {
+      emitMissingValueWarning(
+        {
+          name: node.nodeName,
+          parentName: node.parentNode.nodeName,
+          subElements: missingSubElements
+        },
+        emit
+      );
+    }
+  }
+
+  // When InLine format is used some nodes (i.e <NonLinear>, <Companion>, or <Icon>)
+  // require at least one of the following resources: StaticResource, IFrameResource, HTMLResource
+  if (!isAdInline || !required.inLineResources) {
+    return;
+  }
+
+  const resourceFound = required.inLineResources.some(resource => {
+    return parserUtils.childByName(node, resource);
+  });
+  if (!resourceFound) {
+    emitMissingValueWarning(
+      {
+        name: node.nodeName,
+        parentName: node.parentNode.nodeName,
+        oneOfResources: required.inLineResources
+      },
+      emit
+    );
+  }
+}
+
+/**
+ * Check if a node has sub elements.
+ * @param  {Node} node - The node element.
+ * @returns {Boolean}
+ */
+function hasSubElements(node) {
+  return !node.children || node.children.length !== 0;
```

Comment:
shouldn't it be the opposite? `return node.children && node.children.length`

Reply:
Totally ! I'll will fix it right away!